signature/policy_config.go

```diff
@@ -123,7 +123,7 @@ func (m *policyTransportsMap) UnmarshalJSON(data []byte) error {
 	// So, use a temporary map of pointers-to-slices and convert.
 	tmpMap := map[string]*PolicyTransportScopes{}
 	if err := paranoidUnmarshalJSONObject(data, func(key string) interface{} {
-		transport, ok := transports.KnownTransports[key]
+		transport, ok := registeredtransports.KnownTransports[key]
```

Comment:
> > This will need more work; otherwise an user who imports only a subset of transports would refuse to use a policy.json which defines policies for other transports. …policyTransportScopesWithTransport will need to be taught to deal with a nil transport, and silently accept any key. It seems that requirementsForImageRef does the right thing, or close enough to it, even in such a situation.

> Could you shed some light on how you want this to be done in the code?

- In `policyTransportsMap.UnmarshalJSON`, if `KnownTransports[key]` does not exist, just use `nil`.
- In `policyTransportScopesWithTransport.UnmarshalJSON`, if `m.transport` is `nil` (from the change above), skip the `ValidatePolicyConfigurationScope` call (accept any scope string)
- For `requirementsForImageRef`, add a test that when a transport is not in `KnownTransports` when parsing the policy, but it is in `KnownTransports` at the time of `requirementsForImageRef`, the correct set of scopes is used (more or less re-run the existing test, except creating the policy in a different way).
- Plus as close to 100% test coverage as possible.

I hope that makes some sense.

> > It bothers me a bit that we would accept any typo in transport names, though… perhaps we could add a registeredtransports.AllExistingTransportsRegistered bool = false, with the “all-transports” transports.init() seting it to true, and refusing unknown values if AllExistingTransportsRegistered? The downside is that that would make it impossible to add policies private transports not known to containers/image/transports for a single-application-use.

> Not sure about this, it bothers me that ppl will need to know about this and set/unset this. It could be fine though. 

The idea is that this would work automatically for users who use both all of the known `AllTransports`, and any subset of them. The other side of this is that there would be nothing to manage, no extra work but also no flexibility; if `skopeo` thought it knows about all possible transport names, then it still possible to create a local tool with `mypinkponytransport`, but the system-wide `/etc/containers/policy.json` must not contain a policy for `mypinkponytransport` because `skopeo` would reject that policy file.

Yeah, not sure about this either.